data/recommendation/notify.js

```diff
@@ -1,68 +1,140 @@
+const yo = require('yo-yo');
+
 class Notify {
   start() {
-    this.setup();
+    this.target = this.createBase();
```

Comment:
I bet you can have a unified version of `createBase` and `render`, instead of two separate code paths.